docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,121 @@
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| Name                                                         | Summary                                                                  |
++==============================================================+==========================================================================+
+| `@daml/react <https://www.npmjs.com/package/@daml/react>`_   | React hooks to query/create/exercise DAML contracts                      |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/ledger <https://www.npmjs.com/package/@daml/ledger>`_ | DAML ledger object to connect and directly submit commands to the ledger |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/types  <https://www.npmjs.com/package/@daml/types>`_  | DAML data types                                                          |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
 
-Examples of these are automation and interactive applications that let a user or code react to complex state on the ledger (e.g. the DA Navigator tool).
+You can install any of these libraries by running ``yarn add @daml/react`` in the ``ui`` directory
+of your project. Please explore the ``create-daml-app`` example project to see the usage of these
+libraries.
 
-Which approach to take
-======================
+The interface between the frontend and the backend of your application is defined by your DAML code.
+To make your life easy when interacting with the ledger, the DAML assistant can generate
+corresponding typescript data definitions for the data types declared in DAML.
 
-For all except the simplest applications, we generally recommend the state-driven approach. State-driven applications are easier to reason about when determining correctness, so this makes design and implementation easier.
+.. code-block:: bash
 
-In practice, most applications are actually a mixture of the two styles, with one predominating. It is easier to add some event handling to a state-driven application, so it is better to start with that style.
+  daml codegen ts .daml/dist/<your-project-name.dar> -o daml-ts
 
-Structuring an application
-**************************
+This will create a typescript project under ``daml-ts`` that you can install by changing directory
+to ``daml-ts`` and running ``yarn install`` and then in the root of your project ``yarn workspaces
+run build``. To start the DAML sandbox, run the ``daml-start.sh`` script at the root of the project
+directory of ``create-daml-app``.
```

Comment:
```suggestion
The codegen will generate a typescript project in the ``daml-ts``folder that needs to be connected with the rest of the architecture. To do so navigate to ``daml-ts`` and run ``yarn install`` and then ``yarn workspaces run build``.  We are currently working on making this step easier so that you don't have to do it manually.

Next, to start the DAML sandbox just run the ``daml-start.sh`` script in the ``create-daml-app`` folder.
```

Would try to explain why all the yarn commands for daml-ts project are necessary.

Reply:
added this suggestion with slight modifications.